### PR TITLE
Engine: documents for Engine configuration

### PR DESCRIPTION
**Pull request by remyhaemmerle-da**
plus cosmetic changes.

CHANGELOG_BEGIN
CHANGELOG_END

### Pull Request Checklist

- [X] Read and understand the [contribution guidelines](https://github.com/digital-asset/daml/blob/master/CONTRIBUTING.md)
- [ ] Include appropriate tests
- [ ] Set a descriptive title and thorough description
- [ ] Add a reference to the [issue this PR will solve](https://github.com/digital-asset/daml/issues), if appropriate
- [ ] Include changelog additions in one or more commit message bodies between the `CHANGELOG_BEGIN` and `CHANGELOG_END` tags
- [ ] Normal production system change, include purpose of change in description

NOTE: CI is not automatically run on non-members pull-requests for security
reasons. The reviewer will have to comment with `/AzurePipelines run` to
trigger the build.

### CODE REVIEW

File: daml-lf/engine/src/main/scala/com/digitalasset/daml/lf/engine/EngineConfig.scala [hunk 1]
```diff
@@ -7,23 +7,63 @@ package engine
...
+  *     included in this range
+  *
+  * @param allowedOutputTransactionVersions The range of output
+  *     transactions the engine is allowed to produces. The Engine
+  *     will always use the lowest possible version from this range to
+  *     encode the output transaction, and fails if such version does
+  *     not exist.
+  *
+  * @param stackTraceMode The flag enables the runtime support for
+  *     stack trace.
```
Comment (cocreature) [inline daml-lf/engine/src/main/scala/com/digitalasset/daml/lf/engine/EngineConfig.scala:36]:
```suggestion
  * @param stackTraceMode Enables runtime support for stack traces
```

File: daml-lf/engine/src/main/scala/com/digitalasset/daml/lf/engine/EngineConfig.scala [hunk 2]
```diff
@@ -7,23 +7,63 @@ package engine
...
+  *
+  * @param profileDir The optional specifies the directory where to
+  *     save the output of the DAML scenario profiler. The profiler is
+  *     disable if the option is empty.
```
Comment (cocreature) [inline daml-lf/engine/src/main/scala/com/digitalasset/daml/lf/engine/EngineConfig.scala:40]:
```suggestion
  * @param profileDir Optional directory to save output of the DAML profiler. The profiler is
  *     disabled if this is not set.
```

File: daml-lf/engine/src/main/scala/com/digitalasset/daml/lf/engine/EngineConfig.scala [hunk 3]
```diff
@@ -43,18 +99,16 @@ object EngineConfig {
...
-  val Stable: EngineConfig = new EngineConfig(
-    allowedLanguageVersions = VersionRange(
-      LV(LV.Major.V1, LV.Minor.Stable("6")),
-      LV(LV.Major.V1, LV.Minor.Stable("8")),
-    ),
-    allowedInputTransactionVersions = VersionRange(TV("10"), TV("10")),
-    allowedOutputTransactionVersions = VersionRange(TV("10"), TV("10"))
-  )
+  /**
+    * Recommanded production configuration.
```
Comment (cocreature) [inline daml-lf/engine/src/main/scala/com/digitalasset/daml/lf/engine/EngineConfig.scala:103]:
```suggestion
    * Recommended production configuration.
```

File: daml-lf/transaction/src/main/scala/com/digitalasset/daml/lf/transaction/VersionRange.scala [hunk 4]
```diff
@@ -5,6 +5,20 @@ package com.daml.lf
...
+/**
+  * [[VersionRange]] represents a range of versions of
+  * [[com.daml.lf.language.LanguageVersion]],
+  * [[com.daml.lf.transaction.TransactionVersion]], or
+  * [[com.daml.lf.value.ValueVersion]].
+  *
+  * The order of versions is specified by [[VersionTimeline]].
+  *
+  * @param min the minimal version included in the range.
+  * @param max the maximla version included in the range.
```
Comment (cocreature) [inline daml-lf/transaction/src/main/scala/com/digitalasset/daml/lf/transaction/VersionRange.scala:17]:
```suggestion
  * @param max the maximum version included in the range.
```

**Review by cocreature — APPROVED** [review]
Thanks!

File: daml-lf/engine/src/main/scala/com/digitalasset/daml/lf/engine/EngineConfig.scala [hunk 5]
```diff
@@ -7,24 +7,64 @@ package engine
 import java.nio.file.Path
 
 import com.daml.lf.language.{LanguageVersion => LV}
-import com.daml.lf.transaction.{TransactionVersions, TransactionVersion => TV}
+import com.daml.lf.transaction.{TransactionVersions, TransactionVersion}
+import com.daml.lf.value.ValueVersion
 
+/**
+  * Engine configuration. This describe the  version of language and
```
Comment (hurryabit) [inline daml-lf/engine/src/main/scala/com/digitalasset/daml/lf/engine/EngineConfig.scala:14]:
```suggestion
  * Engine configuration. This describes the version of language and
```

File: daml-lf/engine/src/main/scala/com/digitalasset/daml/lf/engine/EngineConfig.scala [hunk 6]
```diff
@@ -7,24 +7,64 @@ package engine
...
+  * transaction the engine is allowed to read and write. This also
+  * parameterized debugging option.
```
Comment (hurryabit) [inline daml-lf/engine/src/main/scala/com/digitalasset/daml/lf/engine/EngineConfig.scala:16]: I'm not sure I understand what you want to say here. Is this particular class parametrized by a debugging option or are the debugging options parametrized by this class?

File: daml-lf/engine/src/main/scala/com/digitalasset/daml/lf/engine/EngineConfig.scala [hunk 7]
```diff
@@ -7,24 +7,64 @@ package engine
...
+  *     engine is allowed to load.  The engine will crash if it asked
+  *     to load a language version that is not included in this range.
+  *
+  * @param allowedInputTransactionVersions The range of transaction
+  *     version the engine is allowed to load. The engine will crash
+  *     if it is asked to load transaction version that is not
+  *     included in this range
+  *
+  * @param allowedOutputTransactionVersions The range of output
+  *     transactions the engine is allowed to produces. The Engine
```
Comment (hurryabit) [inline daml-lf/engine/src/main/scala/com/digitalasset/daml/lf/engine/EngineConfig.scala:30]:
```suggestion
  *     transactions the engine is allowed to produce. The Engine
```

File: daml-lf/engine/src/main/scala/com/digitalasset/daml/lf/engine/EngineConfig.scala [hunk 8]
```diff
@@ -7,24 +7,64 @@ package engine
...
+  *     will always use the lowest possible version from this range to
+  *     encode the output transaction, and fails if such version does
+  *     not exist.
+  *
+  * @param stackTraceMode The flag enables the runtime support for
+  *     stack trace.
```
Comment (hurryabit) [inline daml-lf/engine/src/main/scala/com/digitalasset/daml/lf/engine/EngineConfig.scala:36]:
```suggestion
  *     stack traces.
```

File: daml-lf/engine/src/main/scala/com/digitalasset/daml/lf/engine/EngineConfig.scala [hunk 9]
```diff
@@ -7,24 +7,64 @@ package engine
...
+  *
+  * @param profileDir The optional specifies the directory where to
+  *     save the output of the DAML scenario profiler. The profiler is
+  *     disable if the option is empty.
```
Comment (hurryabit) [inline daml-lf/engine/src/main/scala/com/digitalasset/daml/lf/engine/EngineConfig.scala:40]:
```suggestion
  *     disabled if the option is empty.
```

File: daml-lf/transaction/src/main/scala/com/digitalasset/daml/lf/transaction/VersionRange.scala (hunk 4 above)
Comment (hurryabit) [inline daml-lf/transaction/src/main/scala/com/digitalasset/daml/lf/transaction/VersionRange.scala:17]:
```suggestion
  * @param max the maximal version included in the range.
```

**Review by hurryabit — APPROVED** [review]
Very good idea to comment this part of our outward facing API. Thank you.

**Pull request merged**